### PR TITLE
add clarification on the acces parameter of the at operator

### PR DESCRIPTION
**Pull request by StevenPuttemans**
To many people are getting stuck at the at operator access parameter for the jump size. Hopefully this clarifies more.

### CODE REVIEW

**Comment by alalek** [comment]
:+1:

File: modules/core/include/opencv2/core/mat.hpp [hunk 1]
```diff
@@ -1702,6 +1702,14 @@ class CV_EXPORTS Mat
             for(int j = 0; j < H.cols; j++)
                 H.at<double>(i,j)=1./(i+j+1);
     @endcode
+
+    Keep in mind that the size identifier used in the at operator cannot be chosen at random. It depends
+    on the image from which you are trying to retrieve the data. The table below gives a better insight in this:
+     - If matrix is of type `CV_8U` or `CV_8S` then use `Mat.at<uchar>(y,x)`.
+     - If matrix is of type `CV_16U` or `CV_16S` then use `Mat.at<ushort>(y,x)`.
+     - If matrix is of type `CV_32S`  or `CV_32F` then use `Mat.at<float>(y,x)`.
```
Comment (mshabunin) [inline modules/core/include/opencv2/core/mat.hpp]: Doesn't `CV_32S` correspond to `int`?

File: modules/core/include/opencv2/core/mat.hpp [hunk 2]
```diff
@@ -1702,6 +1702,14 @@ class CV_EXPORTS Mat
...
+     - If matrix is of type `CV_64FU` then use `Mat.at<double>(y,x)`.
```
Comment (mshabunin) [inline modules/core/include/opencv2/core/mat.hpp]: I guess it should be `CV_64F`, not `CV_64FU`.

**Comment by StevenPuttemans** [comment]
@mshabunin true that should indeed be int, which is a 32 bit jump. That means that ushort and float are both 16 bit?

I will make a PR next monday to apply fixes!

**Comment by Dikay900** [comment]
I think the types are as follow:
CV_8U - uchar
CV_8S - schar
CV_16U - ushort
CV_16S - short
CV_32S - int
CV_32F - float
CV_64F - double

**Comment by StevenPuttemans** [comment]
@Dikay900 correct but basically it defines how much bits to jump in memory. So in my opinion schar and uchar are equal. Same goes for ushort and short. Therefore I wonder if float and int are just both 32 bit jumps and could just be cross used. The range of values it can hold are different but that does not matter here.

**Comment by Dikay900** [comment]
Since these functions are template based types there would be a cast needed there.
Furthermore since you are adding these changes for clarification the user should not struggle using these constants at all after reading this short text.
So i would suggest clarifying every single constant to the user since most of the time they want a specific type and would just trying to cast to their type running straight into other problems. (e.g. getting uchar with original negative values then casting to short)

**Comment by FredFG** [comment]
I ran into this problem yesterday and had to revert to pointers. . Whatever Is behind the curtain, the interface and documentation must be unambiguous. It must be completely clear. 

Frederick Glover P.E.

**Pull request merged**